include/zephyr/sd/sd_spec.h

```diff
@@ -538,6 +557,120 @@ struct sd_csd {
 	/*!< File format [11:10] */
 };
 
+/**
+ * @brief MMC Maximum Frequency
+ *
+ * Max freq in MMC csd
+ */
+enum mmc_csd_freq {
+	MMC_MAXFREQ_100KHZ = 0U << 0U,
+	MMC_MAXFREQ_1MHZ = 1U << 0U,
+	MMC_MAXFREQ_10MHZ = 2U << 0U,
+	MMC_MAXFREQ_100MHZ = 3U << 0U,
+	MMC_MAXFREQ_MULT_10 = 1U << 3U,
+	MMC_MAXFREQ_MULT_12 = 2U << 3U,
+	MMC_MAXFREQ_MULT_13 = 3U << 3U,
+	MMC_MAXFREQ_MULT_15 = 4U << 3U,
+	MMC_MAXFREQ_MULT_20 = 5U << 3U,
+	MMC_MAXFREQ_MULT_26 = 6U << 3U,
+	MMC_MAXFREQ_MULT_30 = 7U << 3U,
+	MMC_MAXFREQ_MULT_35 = 8U << 3U,
+	MMC_MAXFREQ_MULT_40 = 9U << 3U,
+	MMC_MAXFREQ_MULT_45 = 0xAU << 3U,
+	MMC_MAXFREQ_MULT_52 = 0xBU << 3U,
+	MMC_MAXFREQ_MULT_55 = 0xCU << 3U,
+	MMC_MAXFREQ_MULT_60 = 0xDU << 3U,
+	MMC_MAXFREQ_MULT_70 = 0xEU << 3U,
+	MMC_MAXFREQ_MULT_80 = 0xFU << 3u
+};
+
+/**
+ * @brief MMC Timing Modes
+ *
+ * MMC Timing Mode, encoded in EXT_CSD
+ */
+enum mmc_timing_mode {
+	MMC_LEGACY_TIMING = 0U,
+	MMC_HS_TIMING = 1U,
+	MMC_HS200_TIMING = 2U,
+	MMC_HS400_TIMING = 3U
+};
+
+/**
+ * @brief MMC Driver Strengths
+ *
+ * Encoded in EXT_CSD
+ */
+enum mmc_driver_strengths {
+	mmc_driv_type0 = 0U,
+	mmc_driv_type1 = 1U,
+	mmc_driv_type2 = 2U,
+	mmc_driv_type3 = 3U,
+	mmc_driv_type4 = 4U
+};
+
+
+/**
+ * @brief MMC Device Type
+ *
+ * Encoded in EXT_CSD
+ */
+struct mmc_device_type {
+	bool MMC_HS400_DDR_1200MV;
+	bool MMC_HS400_DDR_1800MV;
+	bool MMC_HS200_SDR_1200MV;
+	bool MMC_HS200_SDR_1800MV;
+	bool MMC_HS_DDR_1200MV;
+	bool MMC_HS_DDR_1800MV;
+	bool MMC_HS_52_DV;
+	bool MMC_HS_26_DV;
+};
+
+/**
+ * @brief CSD Revision
+ *
+ * Value of CSD rev in EXT_CSD
+ */
+enum mmc_ext_csd_rev {
+	MMC_5_1 = 8U,
+	MMC_5_0 = 7U,
+	MMC_4_5 = 6U,
+	MMC_4_4 = 5U,
+	MMC_4_3 = 3U,
+	MMC_4_2 = 2U,
+	MMC_4_1 = 1U,
+	MMC_4_0 = 0U
+};
+
+/**
+ * @brief MMC extended card specific data register
+ *
+ * Extended card specific data register.
+ * Contains additional additional data about MMC card.
+ */
+struct mmc_ext_csd {
```

Comment:
Does this need to live here?  Can it be in a header in the subsystem dir?

Reply:
This file is for specification-related definitions, and I am following the precedent Daniel set when he included the sdmmc csd specification struct here as well. The reason mmc and sd specifications are in the same file is because they share a lot of similarities.

Reply:
To add to this, the reason that `sd_spec.h` isn't stored in the subsystem directory is because the `sdhc.h` driver file utilizes specification definitions for some constants, so it includes the `sd_spec.h` header.